web/pdf_document_properties.js

```diff
@@ -29,32 +31,69 @@ class PDFDocumentProperties {
   /**
    * @param {PDFDocumentPropertiesOptions} options
    */
-  constructor(options) {
-    this.overlayName = options.overlayName;
-    this.fields = options.fields;
-    this.container = options.container;
+  constructor({ overlayName, fields, container, closeButton, }) {
+    this.overlayName = overlayName;
+    this.fields = fields;
+    this.container = container;
 
-    this.rawFileSize = 0;
-    this.url = null;
-    this.pdfDocument = null;
+    this._reset();
 
-    // Bind the event listener for the Close button.
-    if (options.closeButton) {
-      options.closeButton.addEventListener('click', this.close.bind(this));
+    if (closeButton) {
+      // Bind the event listener for the Close button.
+      closeButton.addEventListener('click', this.close.bind(this));
     }
-    this._dataAvailableCapability = createPromiseCapability();
-
     OverlayManager.register(this.overlayName, this.container,
                             this.close.bind(this));
   }
 
   /**
    * Open the document properties overlay.
+   *
+   * Note that since the document properties are fetched asynchronously,
+   * the dialog will *only* be updated if it's still open when the data arrives.
    */
   open() {
     Promise.all([OverlayManager.open(this.overlayName),
                  this._dataAvailableCapability.promise]).then(() => {
-      this._getProperties();
+      if (OverlayManager.active !== this.overlayName) {
+        return;
+      }
+      // Get the document properties.
+      this.pdfDocument.getMetadata().then(({ info, metadata, }) => {
+        if (OverlayManager.active !== this.overlayName) {
+          return;
+        }
+        let content = {
```

Comment:
> As I understood it, let is useful for block scoping, but is that necessary here?

It's not technically necessary *here*, but you could also flip the question on its head and ask: "Why would you (generally) use `var` when `let` now is a thing?" :-)
Basically, using `let` can actually help avoid some nasty bugs[1] related to e.g. hoisting of `var`.[2]

There's a lot of useful information in this MDN article: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let.

---
[1] See e.g. https://github.com/mozilla/pdf.js/pull/8355#issue-225492050, which shows how `var` can easily do something other than what you'd expect/intend.
[2] There's obviously cases where you want `var` specifically, but in most cases it doesn't matter, and then I think I'd favour `let` in new code. (The general Firefox frontend code-base also seem to be trending in this general direction, unless I'm not mistaken.)